src/averageAges.js

```diff
@@ -54,11 +73,28 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = onlyWithSon
+    ? people.filter(person => person.sex === 'm' && person.mother !== null)
+    : people.filter(person => person.mother !== null);
+
+  const ageDifference = filteredPeople.map(person => {
+    const mother = people.find(p => p.name === person.mother);
+
+    if (mother && mother.born) {
+      return person.born - mother.born;
```

Comment:
There's no need to check if 'mother.born' is truthy. If the 'mother' exists, 'born' property will always be a number.